### PR TITLE
feat(UI): Get some nice UI improvements from Reservoir

### PR DESCRIPTION
**Pull request by e0ipso**
Task: #60
* [x] Ready for review
* [x] Ready for merge

Addresses many of the features requested in #60. If possible, let's merge as is and let's work on de-scoping the stuff we do not want.

Incognito screenshots are for anonymous users.

### QA

welcome to contenta cms contenta json api 2017-07-10 16-54-01

welcome to contenta cms contenta json api 2017-07-10 16-54-51

people contenta json api 2017-07-10 16-55-30

ploughmans deep pan pizza contenta json api 2017-07-10 16-56-31

contenta json api 2017-07-10 16-57-37

contenta json api 2017-07-10 16-58-44

### CODE REVIEW

File: modules/contenta_enhancements/contenta_enhancements.module [hunk 1]
```diff
@@ -21,3 +48,300 @@ function contenta_enhancements_entity_base_field_info_alter(&$fields, EntityType
...
+
+/**
+ * Implements hook_local_tasks_alter().
+ */
+function contenta_enhancements_local_tasks_alter(&$local_tasks) {
+  $local_tasks['entity.oauth2_client.collection']['title'] = '📱 Clients';
+  $local_tasks['entity.oauth2_client.collection']['base_route'] = 'entity.user.collection';
+  $local_tasks['entity.oauth2_client.collection']['weight'] = 90;
+
+  $local_tasks['entity.oauth2_token.collection']['title'] = '🔑 Tokens';
```
Comment (dawehner) [inline modules/contenta_enhancements/contenta_enhancements.module:68]: Personally I believe if we add icons we should use the ones from material design, let's iterate here :)

File: modules/contenta_enhancements/contenta_enhancements.module [hunk 2]
```diff
@@ -21,3 +48,300 @@ function contenta_enhancements_entity_base_field_info_alter(&$fields, EntityType
...
+        'menu' => $renderable_reports,
+      ],
+      '#weight' => -5,
+    ];
+  }
+  else {
+    $items['reports'] = $empty_item_cacheability;
+  }
+
+  $items['administration']['tab']['#title'] = t('Advanced');
```
Comment (dawehner) [inline modules/contenta_enhancements/contenta_enhancements.module:273]: I love that!

File: modules/contenta_enhancements/contenta_enhancements.module (hunk 1 above)
Comment (e0ipso) [reply modules/contenta_enhancements/contenta_enhancements.module:68]: I'm OK with that. I want to keep it consistent with Reservoir to start talking about abstracting the common bits.

File: modules/contenta_enhancements/contenta_enhancements.module [hunk 3]
```diff
@@ -21,3 +48,300 @@ function contenta_enhancements_entity_base_field_info_alter(&$fields, EntityType
...
+  $items['administration']['#weight'] = 30;
+  if (!$current_user->hasPermission('access administration pages')) {
```
Comment (dawehner) [inline modules/contenta_enhancements/contenta_enhancements.module:275]: The better way would be to use ```$items['administration']['#access'] = $current_user->hasPermission('access administration pages')```

Comment (e0ipso) [reply modules/contenta_enhancements/contenta_enhancements.module:275]: Probably, but they seem to want to have the empty item for cacheability metadata. Surely @wimleers has some input here.

File: modules/contenta_enhancements/src/Controller/ApiAuthInfo.php [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+HTML;
+    }
+
+    $build['info'] = [
+      '#markup' => $html,
+      '#suffix' => '<br><p><small>This uses an OAuth2 password grant to retrieve an access token and a refresh token. Use your favorite library\'s OAuth2 support, or learn how OAuth2 works.</small></p>',
+    ];
+
+    if (\Drupal::request()->getScheme() !== 'https') {
+      drupal_set_message(t('This Contenta instance is not using HTTPS, this is insecure. Do not do this in production!'), 'error');
```
Comment (dawehner) [inline modules/contenta_enhancements/src/Controller/ApiAuthInfo.php:57]: Sounds like a warning for me though.

**Comment by dawehner** [comment]
More improvements will come over time I hope, like better icons :)

I need to wait for travis to merge it :)

File: modules/contenta_enhancements/src/Controller/ApiAuthInfo.php (hunk 4 above)
Comment (e0ipso) [reply modules/contenta_enhancements/src/Controller/ApiAuthInfo.php:57]: Let's make sure to propose this change when consolidation has happened.

**Comment by e0ipso** [comment]
@dawehner travis is green

**Comment by dawehner** [comment]
<3

File: modules/contenta_enhancements/contenta_enhancements.module (hunk 3 above)
Comment (dawehner) [reply modules/contenta_enhancements/contenta_enhancements.module:275]: Well #access would do the same IMHO, it doesn't render anything, but keep all the other bits.

File: modules/contenta_enhancements/src/Controller/ApiAuthInfo.php (hunk 4 above)
Comment (dawehner) [reply modules/contenta_enhancements/src/Controller/ApiAuthInfo.php:57]: fair :)

**Pull request merged**